pkg/controller/secret/secret_controller.go

```diff
@@ -69,25 +66,65 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 	if err != nil {
 		return errors.Wrapf(err, "could not create the controller: %v", ControllerName)
 	}
-	predicateFilter := predicate.Funcs{
-		// get update event only when secret data is changed
-		UpdateFunc: func(e event.UpdateEvent) bool {
-			if string(e.ObjectOld.(*core.Secret).Data["userData"][:]) != string(e.ObjectNew.(*core.Secret).Data["userData"][:]) {
-				return true
-			}
-			return false
-		},
+
+	// Get namespace that the operator is deployed in
```

Comment:
Comment does not match the API you are using.

Reply:
`watchNamespace` maybe?